### DISCUSSION TITLE
Adding external audio sources to the music directory

### DISCUSSION THREAD

**Discussion by geneticdrift** [MusicPlayerDaemon/MPD#2121] [Ideas]
Would there be an interest in adding external audio sources to the music directory as regular files?

The idea is to be able to add an external URI, with attached metadata, to the database, by creating a regular file in the music directory.

An example use-case is for radio stations: instead of each client managing urls and adding them directly to the player queue, it would be possible to put these radio stations as files in the music directory.

It seems to be possible by adding a custom decoder plugin and using the ```Song::target``` .
There's an experimental implementation at this branch: [meta-song](/geneticdrift/MPD/commits/meta-song)

**Comment by MaxKellermann** [comment]
That is already possible easily with a single-line m3u file

**Reply by geneticdrift** [reply]
There are some disadvantages to using playlists for this:
The playlist tracks are not added to the database - can't be filtered or searched, and their metadata are not listed by the list commands.
The metadata is limited and mostly not standardized ([M3U](https://en.wikipedia.org/wiki/M3u#Extended_M3U)).

The idea is to give MPD and its clients more flexibility and uniformity in managing audio sources.

**Comment by bonelifer** [comment]
Have you thought about using mergerfs, instead. No need to add a feature to MPD when it can be done now with mergerfs, or symlinks(which is how I was adding my Gpodder podcast downloads to MPD)

**Reply by geneticdrift** [reply]
Thanks, but this solves a different problem.

I'm looking for a solution to add URLs with meta data to the database like "music" files and have them appear in regular list, find, search commands.
